types/firestore.d.ts

```diff
@@ -97,7 +160,10 @@ declare namespace FirebaseFirestore {
      * toFirestore() must be defined with `Partial<T>`.
      */
     toFirestore(modelObject: T): DocumentData;
-    toFirestore(modelObject: Partial<T>, options: SetOptions): DocumentData;
+    toFirestore(
+      modelObject: NestedPartial<T>,
```

Comment:
I think this is also a breaking change since you are widening the type of a function that has to be implemented by the user.